src/app/qgsnewspatialitelayerdialog.cpp

```diff
@@ -75,20 +78,7 @@ QgsNewSpatialiteLayerDialog::QgsNewSpatialiteLayerDialog( QWidget *parent, Qt::W
   mTypeBox->addItem( QgsApplication::getThemeIcon( QStringLiteral( "/mIconFieldInteger.svg" ) ), tr( "Whole number" ), "integer" );
   mTypeBox->addItem( QgsApplication::getThemeIcon( QStringLiteral( "/mIconFieldFloat.svg" ) ), tr( "Decimal number" ), "real" );
 
-  // Populate the database list from the stored connections
-  QgsSettings settings;
-  settings.beginGroup( QStringLiteral( "SpatiaLite/connections" ) );
-  QStringList keys = settings.childGroups();
-  QStringList::Iterator it = keys.begin();
-  mDatabaseComboBox->clear();
-  while ( it != keys.end() )
-  {
-    // retrieving the SQLite DB name and full path
-    QString text = settings.value( *it + "/sqlitepath", "###unknown###" ).toString();
-    mDatabaseComboBox->addItem( text );
-    ++it;
-  }
-  settings.endGroup();
+  mDatabaseComboBox->setModel( new QgsProviderConnectionModel( QStringLiteral( "spatialite" ), this ) );
```

Comment:
You could just use QgsProviderConnectionComboBox here directly -- that gives you the setConnection method you are missing in createDb